resource-managers/mesos/src/main/scala/org/apache/spark/scheduler/cluster/mesos/MesosCoarseGrainedSchedulerBackend.scala

```diff
@@ -688,14 +689,9 @@ private[spark] class MesosCoarseGrainedSchedulerBackend(
           totalGpusAcquired -= gpus
           gpusByTaskId -= taskId
         }
-        // If it was a failure, mark the slave as failed for blacklisting purposes
         if (TaskState.isFailed(state)) {
           slave.taskFailures += 1
```

Comment:
In the upstream they are removing this count, but here I have kept it for reporting purpose. It will provide the Mesos task failure count.

Reply:
To make sure we're doing everything right here, I suggest adding a unit test and use Mockito to model this situation and verify correct behaviour.